addons/stock/models/stock_lot.py

```diff
@@ -147,6 +152,27 @@ def _compute_delivery_ids(self):
             lot.delivery_ids = delivery_ids_by_lot[lot.id]
             lot.delivery_count = len(lot.delivery_ids)
 
+    def _compute_partner_ids(self):
+        delivery_ids_by_lot = self._find_delivery_ids_by_lot()
+        for lot in self:
+            if len(delivery_ids_by_lot[lot.id]) > 0:
+                lot.partner_ids = self.env['stock.picking'].browse(delivery_ids_by_lot[lot.id]).partner_id
+            else:
+                lot.partner_ids = False
+
+    def _compute_partner_delivery_info(self):
+        if self.env.context.get('active_id'):
+            for lot in self:
+                domain = [('partner_id', '=', self.env.context.get('active_id')), ('move_line_ids.lot_id', '=', lot.id), ('picking_type_code', '=', 'outgoing')]
+                picking_ids = self.env['stock.picking'].search(domain)
+                lot.quantity = sum(
+                        picking.move_ids.move_line_ids.filtered(lambda line: line.lot_id.id == lot.id).quantity
```

Comment:
don't use for loop for this take direct qty using mapped

Reply:
Done.